Flag for comparing file timestamp through git

sergioisidoro opened the pull request:
Attempt at solving #22 

We had this issue because we're running tx client inside a github action -- https://github.com/sergioisidoro/github-transifex-action --, which will clone the repository from scratch. This means the OS timestamp is always newer than the one in transifex.

This PR adds support for fetching the timestamp of a file form the last commit for that file.

TODO: 

- [x] Error handling in case repository is not found
- [x] Testing
- [x] Check if the packaged lib will pick the correct repo of the current directory

deathbird commented on txclib/project.py:
```diff
@@ -394,7 +394,7 @@ def _slug_with_branch(self, resource_slug, branch):
              fetchall=False, fetchsource=False, force=False, skip=False,
              minimum_perc=0, mode=None, pseudo=False, xliff=False, branch=None,
-             parallel=False, no_interactive=False):
+             parallel=False, no_interactive=False, use_git_timestamps=False):
```
`use_git_timestamps` argument should be passed to `_languages_to_pull()` method invocation inside `pull` body.

sergioisidoro commented:
Thank you for the review @deathbird 
I'll try to address these issues this week.

deathbird commented on tests/test_utils.py:
```diff
@@ -364,3 +365,22 @@ def test_project_files_scanning(self):
+        )
+        self.assertIsNotNone(epoch_ts)
+
+    def git_timestamp_is_parsable(self):
```
Rename `git_timestamp_is_parsable()` to `test_git_timestamp_is_parsable()`  - it is not run automatically by tests.

sergioisidoro commented:
Hi @deathbird 

I think I addressed all the issues you mentioned. 

I'm not totally happy with the changes I've made to the `_languages_to_pull` tests (https://github.com/transifex/transifex-client/pull/290/commits/c4f0b9816535abf71d8174a6ea13ab446f6faa6f) but increasing the test coverage with the git workflow might balloon this PR. 

Is there any easy way we could test these? At least the unit tests should cover the sub-methods called by `_languages_to_pull` method.

deathbird commented:
> 
> I'm not totally happy with the changes I've made to the `_languages_to_pull` tests ([c4f0b98](https://github.com/transifex/transifex-client/commit/c4f0b9816535abf71d8174a6ea13ab446f6faa6f)) but increasing the test coverage with the git workflow might balloon this PR.
> 
> Is there any easy way we could test these? At least the unit tests should cover the sub-methods called by `_languages_to_pull` method.

@sergioisidoro thx for addressing the issues.

I have to apologize: Your question above, about the tests, made me delve deeper and realize that `_languages_to_pull()` didn't need to change since it calls `_should_add_translation()` without a local_file argument so the code in `_should_download()` that does the check with the timestamps is never executed. This is logical since  `_should_add_translation()` gets executed to pull new translation files that don't exist locally (so there is no timestamp to check). So my apologies are due for asking you to modify `_languages_to_pull()`. We didn't need to do that in the first place.

So your tests are fine since actually the timestamp check is never executed!

To wrap this up I suggest two approaches:
1) Just squash your commits and Ι proceed to a final approval and we merge.
2) Remove the changes you did in `_languages_to_pull()`, `_should_add_translation()` and dependent tests. I approve, you squash and we merge.

I'm happy with both, though I'd be happier with the 2nd approach :) . However you choose what you want and we proceed.

Thanks in advance.

sergioisidoro commented:
@deathbird Thank you for the review, and no worries. That should be an easy fix. I can just pick out 6cccd85 and c4f0b98 while rebasing the PR. 

I'll re-request review once I'm done 👍

sergioisidoro commented:
Howdy @deathbird.
Rebased the work and picked out 6cccd85 and c4f0b98

The force push diff is here: https://github.com/transifex/transifex-client/compare/ef65adae77de7d28a89e87e6672de740ebad38eb..ccb1af830efd2ea5b4a0b2276224ff57325c9d1d

Please squash when we decide to merge. If we find any issue before merging we can still correct it like I just did, by cherry picking some of the commits out.

<img width="381" alt="Screenshot 2020-06-06 at 19 44 16" src="https://user-images.githubusercontent.com/5014629/83949760-38047680-a82e-11ea-8936-3e4bfd436112.png">
